test/integration/utils.test.js

```diff
@@ -189,4 +189,20 @@ describe(Support.getTestDialectTeaser('Utils'), function() {
       expect(Utils.singularize('status')).to.equal('status');
     });
   });
+
+  describe('formatReferences', function () {
+    ([
+      [{referencesKey: 1}, {references: {model: undefined, key: 1}}],
+      [{references: 'a'}, {references: {model: 'a', key: undefined}}],
+      [{references: 'a', referencesKey: 1}, {references: {model: 'a', key: 1}}],
+      [{references: {model: 1}}, {references: {model: 1}}]
+    ]).forEach(function (test) {
+      var input  = test[0];
+      var output = test[1];
+
+      it('converts ' + JSON.stringify(input) + ' to ' + JSON.stringify(output), function () {
+        expect(Utils.formatReferences(input)).to.deep.equal(output);
+      });
+    });
+  });
```

Comment:
So instead of changing the existing tests, you could just insert new ones to test both. If I counted correct there should be nine occurrences of individual calls to be tested with the old style.